raft/raft.go

```diff
@@ -116,6 +116,9 @@ type Config struct {
 	// used for testing right now.
 	peers []uint64
 
+	// learners can not promote or vote.
```

Comment:
// learners contains the IDs of all leaner nodes (including self if the local node is a leaner) in the raft cluster. 
// learners only receives entries from the leader node. It does not vote or promote itself.